src/notebooklm/_source_upload.py

```
@@ -1,10 +1,531 @@
+                if not t.cancelled() and (exc := t.exception()) is not None:
+                    logger.debug("Background finalize POST failed: %r", exc)
+
+            finalize_task = asyncio.create_task(_do_finalize())
+            finalize_task.add_done_callback(_on_finalize_done)
+            close_wired = True
+            try:
+                await asyncio.shield(finalize_task)
+            except asyncio.CancelledError:
+                if not finalize_started:
+                    finalize_task.cancel()
+                    asyncio.create_task(cancel_upload_session(upload_url, base_url, auth_route))
```

Ruff (0.15.12)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)